src/jsifier.js

```diff
@@ -156,15 +156,33 @@ function JSify(data, functionsOnly) {
             }
           }
         }
-        if (!(MAIN_MODULE || SIDE_MODULE)) {
+        if (!RELOCATABLE) {
           // emit a stub that will fail at runtime
           LibraryManager.library[ident] = new Function("err('missing function: " + ident + "'); abort(-1);");
         } else {
-          var target = (MAIN_MODULE ? '' : 'parent') + "Module['_" + ident + "']";
+          var isGlobalAccessor = ident.startsWith('g$');
+          var realIdent = ident;
+          if (isGlobalAccessor) {
+            realIdent = realIdent.substr(2);
+          }
+
+          var target = (SIDE_MODULE ? 'parent' : '') + "Module['_" + realIdent + "']";
           var assertion = '';
-          if (ASSERTIONS)
-            assertion = 'if (!' + target + ') abort("external function \'' + ident + '\' is missing. perhaps a side module was not linked in? if this function was expected to arrive from a system library, try to build the MAIN_MODULE with EMCC_FORCE_STDLIBS=1 in the environment");';
-          LibraryManager.library[ident] = new Function(assertion + "return " + target + ".apply(null, arguments);");
+          if (ASSERTIONS) {
+            var what = 'function';
+            if (isGlobalAccessor) {
+              what = 'global';
+            }
+            assertion += 'if (!' + target + ') abort("external ' + what + ' \'' + realIdent + '\' is missing. perhaps a side module was not linked in? if this function was expected to arrive from a system library, try to build the MAIN_MODULE with EMCC_FORCE_STDLIBS=1 in the environment");\n';
+
+          }
+          var functionBody;
+          if (isGlobalAccessor) {
+            functionBody = assertion + "return " + target + ";"
```

Comment:
a global accessor is still a function that needs to be called, isn't it? I'm surprised not to see a call here (just `()`) like 2 lines down?

Reply:
For each global address in the system there exist two different accessors:

```
Module['g$foo'] holds a function that returns Module['foo']

Module['foo'] holds that actual address
```

Modules export addresses in the form of foo `foo` but import in the form of `g$foo`.   The `g$foo` functions are created by the JS glue.